operations/configuration_management/NCN_Node_Personalization.md

```diff
@@ -29,62 +66,151 @@ The following procedure describes how to correctly edit the `sat bootprep` files
     - name: ncn-personalization
     ```
 
-1. In the case of an upgrade, execute the following two substeps:
+### 2. Edit CPE and Analytics layers
+
+This section is required if this procedure is being performed as part of a CSM upgrade.
+If that is not the case, then skip ahead to [Update CFS configuration and components](#4-update-cfs-configuration-and-components).
+
+In this section, the `management-bootprep-node-personalization.yaml` file is modified to specify the versions of CPE and Analytics that
+are currently in use on the system. This is necessary because the new versions of CPE and Analytics have not yet been installed at this
+time in the CSM upgrade procedure.
+
+1. (`ncn-m#`) Acquire a copy of the current CPE and Analytics products CFS configuration values already in use.
+
+    There are multiple ways to do this. The following is an example of finding these values by looking at the CFS configuration of a worker NCN on the system.
+
+    1. Find the component name (xname) of one of the NCN worker nodes.
+
+        ```bash
+        WORKER_XNAME=$(cray hsm state components list --role Management --type Node --subrole Worker --format json | jq -r '.Components[0].ID')
+        echo "${WORKER_XNAME}"
+        ```
+
+        Example output:
+
+        ```text
+        x3000c0s7b0n0
+        ```
+
+    1. Find the current desired CFS configuration for this worker.
+
+        ```bash
+        WORKER_CONFIG=$(cray cfs components describe "${WORKER_XNAME}" --format json | jq -r '.desiredConfig')
+        echo "${WORKER_CONFIG}"
+        ```
+
+        Example output:
+
+        ```text
+        ncn-personalization
+        ```
 
-    1. (`ncn-m#`) Acquire a copy of the current CPE and Analytics products CFS configuration values already in use.
-     Obtain the values for the `cloneUrl`, `commit`, and `playbook` lines for each of those two layers in the next step to ensure that they are personalized with the desired configuration for CPE and Analytics.
+    1. Show the CPE and Analytics layers of that configuration.
 
         ```bash
-        cray cfs components describe <ncn-xname> --format json
+        cray cfs configurations describe "${WORKER_CONFIG}" --format json | \
```

Comment:
I'm totally good with it and you are probably aware, but I just wanted to point out that this is different than what we had - we were looking at `cray cfs components describe` in order to see what configuration had last successfully run for the component, whereas this is looking at the desired config layers for the node.

If anything I think this might be better to look at a configuration rather than the state of an individual component.

I don't recall whether we talked any more about the pros and cons, the closest I can find is this slack thread: https://cray.slack.com/archives/CLG4R6SU9/p1665604546333739

Reply:
Yeah, I don't think we discussed it, but my line of thought when I came to that part was exactly like you laid out here. I figured that the configuration was a better source of truth. I had meant to ask you about that but forgot, so I'm glad you spotted it.